docs/search.md

```diff
@@ -125,3 +125,42 @@ $ bundle exec just-the-docs rake search:init
 
 This command creates the `assets/js/zzzz-search-data.json` file that Jekyll uses to create your search index.
 Alternatively, you can create the file manually with [this content]({{ site.github.repository_url }}/blob/main/assets/js/zzzz-search-data.json).
+
+## Custom content for search index
+
+The standard text that is indexed is the page (or post) `.content`, `.title`, and sometimes headers which are already in the `.content`. Other text (Front Matter, data files, etc.) is not indexed. When you want additional text to be indexed, you can customize Just the Docs.
+
+{: .warning }
+> Customizing search indices is an advanced feature that requires Javascript and Liquid knowledge.
+
+1. When your site used a previous version of Just the Docs, you must update the file
+   `assets/js/zzzz-search-data.json` using the [above methods](#generate-search-index-when-used-as-a-gem).
+2. Add a new file named `_includes/lunr/custom-data.json` to your site. Insert your custom Liquid
+   code that reads the page object at `include.page` then generates custom Javascript
+   fields that hold the custom text you want to index. You can verify the fields you output at
+   the generated `assets/js/search-data.json`.
```

Comment:
@mattxwang A very minor meta-nit: "ex" could be read as "excluding", and I don't think we should use it as a short form of "e.g." or "for example".

Reply:
I'm not confident the recent edits to steps 2 and 3 are clear. As now written, it suggests they are disconnected. "Use 2 for liquid". Use "3 for javascript".
But the truth is the coder must use both. That's how I wrote them tightly together, in sequence, and part of the same numbered list.

This is already a very advanced feature. Anyone using it will know what they are doing. I suspect the majority of these advanced users want to liquid->lunr. And the docs should explain that majority scenario.

If you want to document the ability of two different scenarios:
* liquid->javascript->lunr
* javascript->lunr

...then we need to separate sections and two separate numbered lists.

Reply:
> @mattxwang A very minor meta-nit: "ex" could be read as "excluding", and I don't think we should use it as a short form of "e.g." or "for example".
> 
> This seems even more likely to be confusing (at least for European readers) when the colon is omitted, e.g., in "ex front matter, …".

Agreed. "ex" should not be used.  "e.g." or "for example" is the clear and correct usage.